src/v/storage/kvstore.cc

```diff
@@ -453,7 +455,7 @@ ss::future<> kvstore::load_snapshot() {
 
     co_await reader->close();
     if (ex) {
-        throw ex;
+        std::rethrow_exception(ex);
```

Comment:
eek. nice catch. i wonder if clang-tidy catches this stuff.

Reply:
[cppcoreguidelines-slicing](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/slicing.html) isn't smart enough, in my testing :(

Reply:
seems like a reasonable check is that std::exception and derived types are thrown. i just assume that its not common to throw other thigns.

Reply:
caught by https://clang.llvm.org/extra/clang-tidy/checks/hicpp/exception-baseclass.html

Reply:
sweet